code/modules/power/solar.dm

```diff
@@ -238,17 +238,21 @@
 
 		if(istype(W, /obj/item/stack/sheet/glass) || istype(W, /obj/item/stack/sheet/rglass))
 			var/obj/item/stack/sheet/S = W
-			if(S.use(2))
-				glass_type = W.type
-				playsound(src.loc, 'sound/machines/click.ogg', 50, 1)
-				user.visible_message("[user] places the glass on the solar assembly.", "<span class='notice'>You place the glass on the solar assembly.</span>")
-				if(tracker)
-					new /obj/machinery/power/tracker(get_turf(src), src)
+			if(user.can_drop_items())
+				if(user.unEquip(src) && S.use(2))

```

Comment:
It's not going to fail if it passes the first check, the unEquip is just to drop the assembly, as using drop_item would drop the glass.